### PR TITLE
Improvements to Dockerfile

### PR DESCRIPTION
**Pull request by Skyscimitar**
Follow-up of #7.

- Some improvements to the `Dockerfile`
- Remove outdated information in `CONTRIBUTING.md`
- Use GitHub action to install Poetry
- Improve Docker tests

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,19 @@
-# syntax=docker/dockerfile:experimental
-# Enables Docker buildkit caching feature, also requires setting DOCKER_BUILDKIT=1
+# syntax=docker/dockerfile:1.2
```
Comment (Skyscimitar) [inline Dockerfile:1]: This is necessary to be able to use `--mount` in `RUN` commands.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,20 +3,20 @@ on: push
...
-      - name: Set up Poetry
-        run: |
-          curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python - --no-modify-path
-          echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+          python-version: "3.9"
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: "1.1.10"
+          virtualenvs-create: true
```
Comment (tibdex) [inline .github/workflows/test.yml:19]: That's already the default.

```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
 FROM python:3.9-slim AS builder
 
-RUN --mount=type=cache,target=/root/.cache pip install poetry
+RUN --mount=type=cache,target=/root/.cache pip install poetry==1.1.10
 RUN poetry config virtualenvs.create false
 
-COPY poetry.lock .
-COPY pyproject.toml .
+COPY poetry.lock pyproject.toml ./
```
Comment (tibdex) [inline Dockerfile:7]: Why the extra `/`?

```suggestion
COPY poetry.lock pyproject.toml .
```

File: Dockerfile [hunk 4]
```diff
@@ -1,22 +1,18 @@
...
 
 RUN --mount=type=cache,target=/root/.cache poetry install --no-ansi
```
Comment (tibdex) [inline Dockerfile:9]: We should add the `--no-dev` option to make the image smaller.

```suggestion
RUN --mount=type=cache,target=/root/.cache poetry install --no-ansi --no-dev
```

**Review by tibdex — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (Skyscimitar) [reply Dockerfile:9]: I'm trying to get it to work, I'll update the Dockerfile as soon as I do, there seems to be a problem with the `certifi` package although I'm not sure why

**Review by tibdex — APPROVED** [review]

**Pull request merged**